lib/spack/spack/solver/asp.py

```diff
@@ -878,6 +878,13 @@ def pkg_rules(self, pkg, tests):
 
             for value in sorted(values):
                 self.gen.fact(fn.variant_possible_value(pkg.name, name, value))
+                if hasattr(value, 'when'):
```

Comment:
Would it be useful to add an explicit failure here if there is a `Value` without a `when`?

Reply:
I don't think that is possible, given the constructor that has no default value for that argument:
```python
class Value(object):
    """Conditional value that might be used in variants."""
    def __init__(self, value, when):
        self.value = value
        self.when = when
```
We don't validate `when`, but I think it's ok at the moment to skip checks on that.

Reply:
That makes sense.